andela/bp-esa-backend
# #162169823 Create automation data type

#### What does this PR do?
- create `automation` model/migration
- create `slackAutomation` model/migration
- create `emailAutomation` model/migration
- create `freckleAutomation` model/migration
- create association between all automation models
- add index to automation table
- delete model/migration files that are not more needed
- add test to models created
- delete test for models that are not more present

#### Description of Task to be completed?
Create a model(s) and generate migration(s) for the Automation data type.

#### How should this be manually tested?
- Checkout to the branch containing the changes of this Pull Request(`ch-automation-data-type-162169823`)
- You should see the updated models in the models folder
- You should see the updated migrations in the migrations folder
- Run `npm run undo:all:migrations` on the CLI to undo any previous migrations
- Run `npm run migrations` on the CLI to migrate the new models into the database.


#### What are the relevant pivotal tracker stories?
#162169823
